pandas/tests/extension/base/constructors.py

```diff
@@ -83,3 +83,9 @@ def test_pandas_array_dtype(self, data):
         result = pd.array(data, dtype=np.dtype(object))
         expected = pd.arrays.PandasArray(np.asarray(data, dtype=object))
         self.assert_equal(result, expected)
+
+    def test_construct_empty_dataframe(self, dtype):
+        # GH 33623
+        result = pd.DataFrame(columns=["a"], dtype=dtype)
+        expected = pd.DataFrame(data=[], columns=["a"], dtype=dtype)
```

Comment:
```suggestion
        expected = pd.DataFrame({"a": pd.array([], dtype=dtype})
```

This seems a bit safer way to get the expected result.